src/engine/renderer/tr_shade.cpp

```diff
@@ -2806,20 +2802,15 @@ void Tess_StageIteratorGeneric()
 					{
 						if ( r_precomputedLighting->integer || r_vertexLighting->integer )
 						{
-							if ( (tess.surfaceShader->surfaceFlags & SURF_NOLIGHTMAP) &&
-							     !(tess.numSurfaceStages > 0 && tess.surfaceStages[0]->rgbGen == colorGen_t::CGEN_VERTEX) )
-							{
-								Render_lightMapping( stage, false, false, true );
-							}
-							else if ( !r_vertexLighting->integer && tess.lightmapNum >= 0 && tess.lightmapNum <= tr.lightmaps.currentElements )
+							if ( !r_vertexLighting->integer && tess.lightmapNum >= 0 && tess.lightmapNum <= tr.lightmaps.currentElements )
 							{
 								if ( tr.worldDeluxeMapping && r_normalMapping->integer )
 								{
-									Render_lightMapping( stage, false, true, false );
+									Render_lightMapping( stage, false, true );
 								}
 								else
 								{
-									Render_lightMapping( stage, false, false, false );
+									Render_lightMapping( stage, false, false );
 								}
 							}
 							else if ( backEnd.currentEntity != &tr.worldEntity )
```

Comment:
This can be reached if r_vertexLighting = 0 and tess.lightmapNum is invalid which doesn't seem right.

Reply:
is it something introduced by my change or something that was already there?

Reply:
Already there. Leave a TODO?

Reply:
I just added a `FIXME`